src/components/ec-table-filter/ec-table-filter.spec.js

```diff
@@ -0,0 +1,125 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import EcTableFilter from './ec-table-filter.vue';
+import EcSyncMultipleValuesFilter from '../ec-sync-multiple-values-filter';
+import EcDateRangeFilter from '../ec-date-range-filter';
+import { withMockedConsole } from '../../../tests/utils/console';
+
+const filters = [{
+  label: 'Payment status',
+  name: 'paymentStatus',
+  component: EcSyncMultipleValuesFilter,
+  items: [{ text: 'Paid', value: 'paid' }, { text: 'Cancelled', value: 'canceled' }, { text: 'Overdue', value: 'overdue' }],
+  isSearchable: false,
+  isSelectAll: false,
+  selectAllFiltersText: '',
+}, {
+  label: 'Fee type',
+  name: 'feeType',
+  component: EcSyncMultipleValuesFilter,
+  items: [{ text: 'Invoiced', value: 'invoiced' }, { text: 'Other type', value: 'other type' }],
+  isSearchable: false,
+  isSelectAll: false,
+  selectAllFiltersText: '',
+}, {
+  label: 'Due date',
+  name: 'dueDate',
+  component: EcDateRangeFilter,
+  fromLabelText: 'From',
+  toLabelText: 'To',
+  clearText: 'Clear dates',
+}];
+
+const value = { feeType: [{ text: 'Invoiced', value: 'invoiced' }] };
+
+function mountEcTableFilter(props, mountOpts) {
+  return mount(EcTableFilter, {
+    stubs: { EcPopover: true },
+    propsData: {
+      ...props,
+    },
+    ...mountOpts,
+  });
+}
+
+function mountEcTableFilterAsTemplate(template, props, wrapperComponentOpts, mountOpts) {
+  const localVue = createLocalVue();
+
+  const Component = localVue.extend({
+    components: { EcTableFilter },
+    template,
+    ...wrapperComponentOpts,
+  });
+
+  return mount(Component, {
+    localVue,
+    stubs: { EcPopover: true },
+    propsData: { ...props },
+    ...mountOpts,
+  });
+}
+
+describe('EcTableFilter', () => {
+  it('should render correctly if all the required props are passed', () => {
+    const wrapper = mountEcTableFilter({ filters });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should throw an error if no filters prop is given', () => {
+    withMockedConsole((errorSpy) => {
+      mountEcTableFilter();
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Missing required prop: "filters"');
+    });
+  });
+
+  it('should render with pre-selected filters if the value prop is passed', () => {
+    const wrapper = mountEcTableFilter({ value, filters });
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should hide the clear filters button if there isn\'t any preselected filter', () => {
```

Comment:
I need to add one more test to check if the clear filters button is hidden if we manually de select the options